### PR TITLE
Check SUSEConnect output on read-only filesystem

### PR DESCRIPTION
**Pull request by SergioAtSUSE**
- Related ticket: https://progress.opensuse.org/issues/41258
- Verification run: Textmode role: http://slindomansilla-vm.qa.suse.de/tests/1056

### CODE REVIEW

File: tests/console/suseconnect_scc.pm [hunk 1]
```diff
@@ -34,7 +34,9 @@ sub run {
 
     select_console 'root-console';
     assert_script_run "SUSEConnect --url $scc_url -r $reg_code";
-    assert_script_run 'SUSEConnect --list-extensions';
+    my $expected_suseconnect_output = 'SUSEConnect -p';
+    $expected_suseconnect_output = 'transactional-update register -p' if (get_var('SYSTEM_ROLE') =~ /microos|serverro/);
```
Comment (okurz) [inline tests/console/suseconnect_scc.pm]: hm, how about a simple conditional assignment in the declaration line?:

```perl
my $transactional = get_var('SYSTEM_ROLE') =~ /microos|serverro/
my $expected_suseconnect_output = $transactional ? 'transactional-update register -p' : 'SUSEConnect -p';
```

File: tests/console/suseconnect_scc.pm [hunk 2]
```diff
@@ -34,7 +34,9 @@ sub run {
...
+    validate_script_output('SUSEConnect --list-extensions', sub { $expected_suseconnect_output });
```
Comment (okurz) [inline tests/console/suseconnect_scc.pm]: I recommend to keep the module "suseconnect_scc" focused on registering modules which is needed as a pre-requisite for tests and instead move the actual testing of SUSEConnect to another, explicit test-module

File: tests/console/suseconnect_scc.pm (hunk 1 above)
Comment (SergioAtSUSE) [reply tests/console/suseconnect_scc.pm]: Actually, with your suggestion I just realize that I still need here a function "is_transactional_system".

**Comment by SergioAtSUSE** [comment]
Closing, as this will be done on https://progress.opensuse.org/issues/27853

**Pull request closed without merging**